shopfloor_reception_packaging_dimension/services/reception.py

```diff
@@ -0,0 +1,194 @@
+                [("barcode", "=", False)],
+            ]
+        )
+
+    def _get_next_packaging_to_set_dimension(self, product, previous_packaging=None):
+        """Return for a product the next packaging needing dimension to be set."""
+        next_packaging_id = previous_packaging.id + 1 if previous_packaging else 0
+        domain_dimension = self._get_domain_packaging_needs_dimension()
+        domain_packaging_id = [
+            ("product_id", "=", product.id),
+            ("id", ">=", next_packaging_id),
+        ]
+        domain = expression.AND([domain_packaging_id, domain_dimension])
+        return self.env["product.packaging"].search(domain, order="id", limit=1)
+
+    def _response_for_set_packaging_dimension(
+        self, picking, line, packaging, message=None
+    ):
+        return self._response(
+            next_state="set_packaging_dimension",
+            data={
+                "picking": self.data.picking(picking),
+                "selected_move_line": self.data.move_line(line),
+                "packaging": self.data_detail.packaging_detail(packaging),
+            },
+            message=message,
+        )
+
+    def set_packaging_dimension(
+        self, picking_id, selected_line_id, packaging_id, cancel=False, **kwargs
+    ):
+        """Set the dimension on a product packaging.
+
+        If the user cancel the dimension update we still propose the next
+        possible packgaging.
+
+        Transitions:
+            - set_packaging_dimension: if more packaging needs dimension
+            - set_quantity: otherwise
+        """
+        picking = self.env["stock.picking"].browse(picking_id)
+        selected_line = self.env["stock.move.line"].browse(selected_line_id)
+        packaging = self.env["product.packaging"].sudo().browse(packaging_id)
+        message = None
+        next_packaging = None
+        if not packaging:
+            message = self.msg_store.record_not_found()
+        elif not cancel and self._check_dimension_to_update(kwargs):
+            self._update_packaging_dimension(packaging, kwargs)
+            message = self.msg_store.packaging_dimension_updated(packaging)
+        if packaging:
+            next_packaging = self._get_next_packaging_to_set_dimension(
+                selected_line.product_id, packaging
+            )
+        if not next_packaging:
+            return self._response_for_set_quantity(
+                picking, selected_line, message=message
+            )
+        return self._response_for_set_packaging_dimension(
+            picking, selected_line, next_packaging, message=message
+        )
+
+    def _check_dimension_to_update(self, dimensions):
+        """Return True if there is any dimension that needs to be updated on the packaging."""
+        return any([value is not None for key, value in dimensions.items()])
+
+    def _get_dimension_fields_conversion_map(self):
+        return {"weight": "max_weight", "length": "packaging_length"}
```

Comment:
mapping for weight is not needed, because we will get max_weight from js
```suggestion
        return {"length": "packaging_length"}
```

Reply:
Fixed